Errors with matplotib 3.6.0

### What happened?

On Mac unprovisioned builds, matplotlib was updated from 3.5.3 to 3.6.0 last night. This caused the `//common/proto:call_python_test` to fail:

```
Plotting a simple 3D surface
Traceback (most recent call last):
  File "/Users/bigsur/workspace/mac-big-sur-unprovisioned-clang-bazel-nightly-release/_bazel_bigsur/bec0ae6dc20cf0d6dbe6178cb16ef78f/sandbox/darwin-sandbox/4191/execroot/drake/bazel-out/darwin-opt/bin/common/proto/call_python_test.runfiles/drake/common/proto/call_python_client.py", line 371, in _execute_message
    self._execute_message_impl(msg)
  File "/Users/bigsur/workspace/mac-big-sur-unprovisioned-clang-bazel-nightly-release/_bazel_bigsur/bec0ae6dc20cf0d6dbe6178cb16ef78f/sandbox/darwin-sandbox/4191/execroot/drake/bazel-out/darwin-opt/bin/common/proto/call_python_test.runfiles/drake/common/proto/call_python_client.py", line 426, in _execute_message_impl
    out = eval(function_name + "(*_tmp_args, **_tmp_kwargs)",
  File "<string>", line 1, in <module>
  File "/Users/bigsur/workspace/mac-big-sur-unprovisioned-clang-bazel-nightly-release/_bazel_bigsur/bec0ae6dc20cf0d6dbe6178cb16ef78f/sandbox/darwin-sandbox/4191/execroot/drake/bazel-out/darwin-opt/bin/common/proto/call_python_test.runfiles/drake/common/proto/call_python_client.py", line 258, in surf
    ax = fig.gca(projection='3d')
TypeError: FigureBase.gca() got an unexpected keyword argument 'projection'
  Continuing (no --stop_on_error)

....
======================================================================
FAIL [8.892s]: test_basic (call_python_test.TestCallPython)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bigsur/workspace/mac-big-sur-unprovisioned-clang-bazel-nightly-release/_bazel_bigsur/bec0ae6dc20cf0d6dbe6178cb16ef78f/sandbox/darwin-sandbox/4191/execroot/drake/bazel-out/darwin-opt/bin/common/proto/call_python_test.runfiles/drake/common/proto/test/call_python_test.py", line 113, in test_basic
    self.run_server_and_client(with_error)
  File "/Users/bigsur/workspace/mac-big-sur-unprovisioned-clang-bazel-nightly-release/_bazel_bigsur/bec0ae6dc20cf0d6dbe6178cb16ef78f/sandbox/darwin-sandbox/4191/execroot/drake/bazel-out/darwin-opt/bin/common/proto/call_python_test.runfiles/drake/common/proto/test/call_python_test.py", line 100, in run_server_and_client
    self.assertEqual(client_status, int(with_error))
AssertionError: 1 != 0
```

See slack thread [here](https://drakedevelopers.slack.com/archives/C270MN28G/p1663335286351369)

### Version

_No response_

### What operating system are you using?

_No response_

### What installation option are you using?

_No response_

### Relevant log output

_No response_

--- a/common/proto/call_python_client.py
+++ b/common/proto/call_python_client.py
@@ -224,8 +224,7 @@ def default_globals():
 
     def box(bmin, bmax, rstride=1, cstride=1, **kwargs):
         """Plots a box bmin[i] <= x[i] <= bmax[i] for i < 3."""
-        fig = plt.gcf()
-        ax = fig.gca(projection='3d')
+        ax = plt.subplot(projection='3d')
         u = np.linspace(1, 9, 5) * np.pi / 4
         U, V = np.meshgrid(u, u)
         cx, cy, cz = (bmax + bmin) / 2
@@ -237,14 +236,12 @@ def default_globals():
 
     def plot3(x, y, z, **kwargs):
         """Plots a 3d line plot."""
-        fig = plt.gcf()
-        ax = fig.gca(projection='3d')
+        ax = plt.subplot(projection='3d')
         ax.plot(x, y, z, **kwargs)
 
     def sphere(n, rstride=1, cstride=1, **kwargs):
         """Plots a sphere."""
-        fig = plt.gcf()
-        ax = fig.gca(projection='3d')
+        ax = plt.subplot(projection='3d')
         u = np.linspace(0, np.pi, n)
         v = np.linspace(0, 2 * np.pi, n)
         X = np.outer(np.sin(u), np.sin(v))
@@ -254,8 +251,7 @@ def default_globals():
 
     def surf(x, y, Z, rstride=1, cstride=1, **kwargs):
         """Plots a 3d surface."""
-        fig = plt.gcf()
-        ax = fig.gca(projection='3d')
+        ax = plt.subplot(projection='3d')
         X, Y = np.meshgrid(x, y)
         ax.plot_surface(X, Y, Z, rstride=rstride, cstride=cstride, **kwargs)
 

